rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -854,10 +875,15 @@ def _predict_label(
             self.component_config[LOSS_TYPE] == SOFTMAX
             and self.component_config[RANKING_LENGTH] > 0
         ):
-            message_sim = train_utils.normalize(
+            message_sim = train_utils.sort_and_rank(
                 message_sim, self.component_config[RANKING_LENGTH]
             )
 
+            if self.component_config[RELATIVE_CONFIDENCE]:
+                # Normalize the values if returned probabilities are from
+                # softmax(hence relative to each other).
+                message_sim = train_utils.normalize(message_sim)
```

Comment:
similar here, I don't think cosine should be normalized

Reply:
Added a check 👍